### DISCUSSION TITLE
Use BelongsToMany in getTableQuery

### DISCUSSION THREAD

**Discussion by obrunsmann** [filamentphp/filament#6710] [Help] [answered]
I am building a table to list team's users. I need to include my pivot data in the filament table:

```php
public function users()
{
    return $this->belongsToMany(User::class, Membership::class)
        ->withPivot(
            'employee_number',
// ...
        );
}
```

Using `return auth()->user()->currentTeam->users();` does not work:

```
MemberList::applyFiltersToTableQuery(): Argument #1 ($query) must be of type Illuminate\Database\Eloquent\Builder, Illuminate\Database\Eloquent\Relations\BelongsToMany given, called in /var/www/html/vendor/filament/tables/src/Concerns/HasRecords.php on line 31
```

Any chance to use the relationship as data source in the table? (Standalone filament table installation)

**Comment by danharrin** [comment, marked as answer]
Try `return auth()->user()->currentTeam->users()->toQuery()`

However, since this is a relationship table, you probably want to go about it differently, so pivot columns are handled better.

Add `HasRelationshipTable` as an interface after `HasTable`. Then define `getRelationship()` and `getInverseRelationship()`, and remove `getTableQuery()`. That should allow pivot data to work correctly.

**Reply by obrunsmann** [reply]
Awesome, that's exactly what I was searching for. Looks like `HasRelationshipTable` is missing in docs... :(

**Reply by chrismartinez99** [reply]
@obrunsmann  would you be able to share your how you implemented @danharrin  answer? I have a similar situation and could use the help.

Thanks!

**Reply by HRankit** [reply]
@obrunsmann same here. I too have the same question.

**Reply by obrunsmann** [reply]
It works just like with my relation now:


```php
    public function users()
    {
        return $this->belongsToMany(Jetstream::userModel(), Jetstream::membershipModel())
            ->withPivot(
                'employee_number',
            )
            ->withTimestamps()
            ->as('membership');
    }
```

Table

```php
    public function table(Table $table): Table
    {
        return $table
            ->query(fn () => auth()->user()->currentTeam->users())
```
